tests/templates/test_integration.py

```diff
@@ -754,30 +505,30 @@ def circuit(*diffable):
             template(**dict)
             return qml.expval(qml.Identity(0))
 
-        # Check gradients in numpy interface
-        if interface == 'numpy':
-            grd = qml.grad(circuit, argnum=argnum)
-            grd(*diffable)
-
-        # Check gradients in torch interface
-        if interface == 'torch':
-            for a in argnum:
-                diffable[a] = TorchVariable(diffable[a], requires_grad=True)
-            res = circuit(*diffable)
-            res.backward()
-            for a in argnum:
-                diffable[a].grad.numpy()
-
-        # Check gradients in tf interface
-        if interface == 'tf':
-            grad_inpts = [diffable[a] for a in argnum]
-            with tf.GradientTape() as tape:
-                loss = circuit(*diffable)
-                tape.gradient(loss, grad_inpts)
-
-    @pytest.mark.parametrize("template, diffable, nondiffable, argnum", CV_DIFFABLE_NONDIFFABLE_ARGNUM)
+        # Do gradient check for every differentiable argument
+        for argnum in range(len(diffable)):
+
+            # Check gradients in numpy interface
+            if interface == 'numpy':
+                grd = qml.grad(circuit, argnum=[argnum])
+                grd(*diffable)
```

Comment:
I printed the gradient and got zeros everywhere, which is expected if we're taking the expectation value w.r.t. the identity. But I also changed the observable to `PauliZ` and still got all zeros. Is that right? :thinking:

I'm just wondering if there's something stronger that we can say, e.g., that at least one gradient is nonzero?

Reply:
Depends on which case you check - for me AngleEmbedding and numpy were giving nice non-zero gradients...